File: src/dart/armstrong_number.dart
import 'dart:math';

bool isArmstrongNumber(int num) {
  if (num < 0) return false;
  var sum = 0;
  num.toString()
      .split('')
      .forEach((ch) => sum += pow(int.parse(ch), num.toString().length) as int);
  return sum == num;
}

File: src/javascript/armstrongNumber.js
const isArmstrongNumber = (num) => {
    let sum = 0;
    let temp = num;

    while (temp > 0) {
        sum += Math.pow(temp%10,3);
        temp = Math.floor(temp/10);
    }

    return sum === num;
}

export default isArmstrongNumber;